lib-src/rpc2/rpc2-src/pack_helper.c

```diff
@@ -0,0 +1,383 @@
+/* BLURB lgpl
+
+                           Coda File System
+                              Release 5
+
+          Copyright (c) 1987-2016 Carnegie Mellon University
+                  Additional copyrights listed below
+
+This  code  is  distributed "AS IS" without warranty of any kind under
+the  terms of the  GNU  Library General Public Licence  Version 2,  as
+shown in the file LICENSE. The technical and financial contributors to
+Coda are listed in the file CREDITS.
+
+#*/
+
+
+#include <rpc2/pack_helper.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+int unpack_int(BUFFER *buf, RPC2_Integer *ptr)
+{
+    if (buf->buffer + 4 > buf->eob)
+        return -1;
+    *ptr = ntohl(*(RPC2_Integer *)(buf->buffer));
+    buf->buffer += 4;
+    return 0;
+}
+
+
+int unpack_unsigned(BUFFER *buf, RPC2_Unsigned *ptr)
+{
+    if (buf->buffer + 4 > buf->eob)
+        return -1;
+    *ptr = ntohl(*(RPC2_Unsigned*)(buf->buffer));
+    buf->buffer += 4;
+    return 0;
+}
+
+
+int unpack_double(BUFFER *buf, RPC2_Double *ptr)
+{
+    if (buf->buffer + 8 > buf->eob)
+        return -1;
+    *ptr = *(RPC2_Double *)(buf->buffer);
+    buf->buffer += 8;
+    return 0;
+}
+
+
+int unpack_bound_bytes(BUFFER *buf, unsigned char *ptr, RPC2_Unsigned len)
+{
+    if (buf->buffer + len > buf->eob)
+        return -1;
+    memcpy(ptr, buf->buffer, len);
+    buf->buffer += _PAD(len);
+    return 0;
+}
+
+int unpack_unbound_bytes(BUFFER *buf, unsigned char *ptr)
+{
+    if (buf->buffer + 4 > buf->eob)
```

Comment:
Actually we are only getting a single byte, so we can test for `buffer + 1 > buf->eob`, and use `buffer += _PAD(1);`, I think.